### PR TITLE
chore(logging): replace zap with log/slog

### PR DESCRIPTION
**Pull request by daveworth**
According to our [Technical Standards for Golang](https://coda.io/d/_dfvozy2Mzwt/Golang_suU9hPBz#_luMQnbXm) (which did not exist when this was created) we use Golang's stdlib's `log/slog` package instead of Uber's Zap package (which used to be my preferred package as well).

### CODE REVIEW

File: pkg/scan/local_scan_test.go [hunk 1]
```diff
@@ -3,28 +3,22 @@ package scan
...
-
-func (m *mockLogger) Debug(msg string, fields ...interface{})  {}
-func (m *mockLogger) Info(msg string, fields ...interface{})   {}
-func (m *mockLogger) Warn(msg string, fields ...interface{})   {}
-func (m *mockLogger) Error(msg string, fields ...interface{})  {}
-func (m *mockLogger) Fatalf(msg string, fields ...interface{}) {}
-
 func TestNewLocalPackageScanner(t *testing.T) {
-	logger := &mockLogger{}
+	logger := slog.New(slog.NewJSONHandler(os.Stdout, nil))
```
Comment (daveworth) [inline pkg/scan/local_scan_test.go:17]: This _could_ be unfortunately noisy - I'm happy to configure it differently but there's no reason for the `mockLogger` at this point.

**Comment by daveworth** [comment]
Note - I am actively and explicitly ignoring the codecov failure since it has been red for some time and the team is working to improve.

File: pkg/scan/local_scan_test.go (hunk 1 above)
Comment (partkyle) [reply pkg/scan/local_scan_test.go:17]: it should be fine, and we can come up with a silencing strategy if it is too noisy.

File: cmd/store/main.go [hunk 2]
```diff
@@ -125,25 +123,22 @@ func runStoreScanner(cmd *cobra.Command, _ []string) error {
...
-	manager, err := db.NewGormScanManager(config.DBConn)
+	manager, err := db.NewGormScanManager(config.DBConn, logger)
 	if err != nil {
 		return fmt.Errorf("error initializing GormScanManager: %w", err)
 	}
-	remoteScanner, ok := scanner.(*scan.Scanner)
-	if !ok {
-		return fmt.Errorf("error creating remote package scanner")
-	}
-	return runStoreScannerWithDeps(ctx, cmd, logInstance, remoteScanner, manager, config)
```
[the remark this answers is not in the document]
Comment (partkyle) [reply cmd/store/main.go:139]: I think that type asserting is vestigial, so I'm glad to see it removed.

**Review by partkyle — APPROVED** [review]

**Pull request merged**